core/io/http_client.cpp

```diff
@@ -489,6 +490,8 @@ Error HTTPClient::poll() {
 		case STATUS_CANT_RESOLVE: {
 			return ERR_CANT_RESOLVE;
 		} break;
+		case STATUS_BODY: {
+		} break;
```

Comment:
```
core/io/http_client.cpp:288:9: warning: enumeration value 'STATUS_BODY' not handled in switch [-Wswitch]
core/io/http_client.cpp:288:9: warning: enumeration value 'STATUS_SSL_HANDSHAKE_ERROR' not handled in switch [-Wswitch]
```
@Faless

Reply:
I would consider `STATUS_BODY` the same case as `STATUS_CONNECTED`. So the socket gets polled, and we can detect disconnect without having to call `read_response_body_chunk` (although that is unlikely, as the user should fetch the body as soon as it arrives). What do you think?
`STATUS_SSL_HANDSHAKE_ERROR` is good,

Reply:
Sounds good, I'll do that change once I have some more edits to make based on other reviews.